src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -100,6 +100,13 @@ public static function createResourceForClass(\ReflectionClass $reflectionClass)
      */
     protected function processValue($value, $isRoot = false)
     {
+        if ($value instanceof Reference && !$this->container->has($id = (string) $value)) {
+            if (preg_match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+(?:\\\\[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+)++$/', $id) && $reflectionClass = $this->container->getReflectionClass($id, true)) {
```

Comment:
We use this regex in other places as well, don't we? What about creating a helper method then? Would increase readability as well as it allows to fix issues only in one place.